src/public/modules/forms/base/directives/submit-form.directive.js

```diff
@@ -424,11 +425,11 @@ function submitFormDirective(
 
       // Create a transaction if there are fields to be verified and the form is intended for submission
       if (!scope.disableSubmitButton) {
-        Verification.createTransaction({ formId: scope.form._id }).then(
-          ({ transactionId }) => {
-            if (transactionId) scope.transactionId = transactionId
-          },
-        )
+        createTransactionForForm(scope.form._id).then((res) => {
```

Comment:
do you need to wrap this in `$q.when`?

Reply:
no rendering depends on this function so didn't use it

Reply:
`transactionId` no meh?
```
createTransactionForForm(scope.form._id).then((res) => {
  if (res.transactionId) {
    scope.transactionId = res.transactionId
  }
})
```